packages/flutter_tools/lib/src/platform_plugins.dart

```diff
@@ -532,6 +535,78 @@ class LinuxPlugin extends PluginPlatform implements NativeOrDartPlugin {
   }
 }
 
+class CustomEmbedderPlugin extends PluginPlatform implements NativeOrDartPlugin {
```

Comment:
Related to my comments on the design document: I don't think that this should implement `NativeOrDartPlugin`, because it's a specific set of implementation keys, and keys are 100% platform-dependent. An alternate web embedder, for instance, would definitely not be a `NativeOrDartPlugin`.